### ISSUE TITLE
When compile env_logger from git, Rustc panicked.

### ISSUE THREAD

**Issue by mindcat** [rust-lang/rust#23075]
```bash
╭─mindcat@mindcat-linux-pc ~/workspace/nekoneko  ‹master*› 
╰─➤  RUST_BACKTRACE=1 RUST_LOG=nekoneko cargo build
   Compiling env_logger v0.2.2 (https://github.com/rust-lang/log#e78b7368)
/home/rustbuild/src/rust-buildbot/slave/nightly-dist-rustc-linux/build/src/libstd/sync/mutex.rs:177:37: 180:2 error: internal compiler error: debuginfo: Could not find scope info for node NodeExpr(Expr { id: 8309, node: ExprStruct(Path { span: Span { lo: BytePos(4566650), hi: BytePos(4566661), expn_id: ExpnId(4294967295) }, global: false, segments: [PathSegment { identifier: StaticMutex#0, parameters: AngleBracketedParameters(AngleBracketedParameterData { lifetimes: [], types: [], bindings: [] }) }] }, [Field { ident: Spanned { node: lock#0, span: Span { lo: BytePos(1770102), hi: BytePos(1770106), expn_id: ExpnId(4294967295) } }, expr: Expr { id: 8310, node: ExprPath(None, Path { span: Span { lo: BytePos(4566674), hi: BytePos(4566689), expn_id: ExpnId(4294967295) }, global: false, segments: [PathSegment { identifier: sys#0, parameters: AngleBracketedParameters(AngleBracketedParameterData { lifetimes: [], types: [], bindings: [] }) }, PathSegment { identifier: MUTEX_INIT#0, parameters: AngleBracketedParameters(AngleBracketedParameterData { lifetimes: [], types: [], bindings: [] }) }] }), span: Span { lo: BytePos(4566674), hi: BytePos(4566689), expn_id: ExpnId(4294967295) } }, span: Span { lo: BytePos(4566668), hi: BytePos(4566689), expn_id: ExpnId(4294967295) } }, Field { ident: Spanned { node: poison#0, span: Span { lo: BytePos(1770129), hi: BytePos(1770135), expn_id: ExpnId(4294967295) } }, expr: Expr { id: 8311, node: ExprPath(None, Path { span: Span { lo: BytePos(4566703), hi: BytePos(4566720), expn_id: ExpnId(4294967295) }, global: false, segments: [PathSegment { identifier: poison#0, parameters: AngleBracketedParameters(AngleBracketedParameterData { lifetimes: [], types: [], bindings: [] }) }, PathSegment { identifier: FLAG_INIT#0, parameters: AngleBracketedParameters(AngleBracketedParameterData { lifetimes: [], types: [], bindings: [] }) }] }), span: Span { lo: BytePos(4566703), hi: BytePos(4566720), expn_id: ExpnId(4294967295) } }, span: Span { lo: BytePos(4566695), hi: BytePos(4566720), expn_id: ExpnId(4294967295) } }], None), span: Span { lo: BytePos(4566650), hi: BytePos(4566723), expn_id: ExpnId(4294967295) } })
note: the compiler unexpectedly panicked. this is a bug.
note: we would appreciate a bug report: https://github.com/rust-lang/rust/blob/master/CONTRIBUTING.md#bug-reports
note: run with `RUST_BACKTRACE=1` for a backtrace
thread 'rustc' panicked at 'Box<Any>', /home/rustbuild/src/rust-buildbot/slave/nightly-dist-rustc-linux/build/src/libsyntax/diagnostic.rs:129

stack backtrace:
   1:     0x7fa61c039f1f - sys::backtrace::write::hf79a3da4fdecb8a0OBA
   2:     0x7fa61c064c32 - panicking::on_panic::h9f64f4c69e19f194hHJ
   3:     0x7fa61bf99eda - rt::unwind::begin_unwind_inner::h37f4496c980fe936knJ
   4:     0x7fa6193ecabd - rt::unwind::begin_unwind::h8320268356453106285
   5:     0x7fa6193eca63 - diagnostic::SpanHandler::span_bug::h83c8af232eaba6a9h0D
   6:     0x7fa619cc8ca3 - session::Session::span_bug::h857b2c7ae23c9286ISp
   7:     0x7fa61b7e680c - trans::debuginfo::scope_metadata::hac54dfdbdcd04cd9SjE
   8:     0x7fa61b6f9408 - trans::debuginfo::set_source_location::h1067a74086ed9dd48MD
   9:     0x7fa61b6ade42 - trans::expr::trans_into::h95c6d2681fdd2548znh
  10:     0x7fa61b6ae109 - trans::expr::trans_into::h95c6d2681fdd2548znh
  11:     0x7fa61b71067f - trans::expr::trans_uniq_expr::h5f082eea62818f84ukj
  12:     0x7fa61b7112ef - trans::expr::trans_unary::h4412379888608420Jgj
  13:     0x7fa61b6fb80e - trans::expr::trans_unadjusted::hfd3a5e1b5cbe37d5z4h
  14:     0x7fa61b6ae417 - trans::expr::trans_into::h95c6d2681fdd2548znh
  15:     0x7fa61b72e6f9 - trans::expr::trans_adt::h1af69b9b4e52152aO6i
  16:     0x7fa61b7311af - trans::expr::trans_struct::closure.42069
  17:     0x7fa61b71b016 - trans::expr::trans_struct::hcae8f9103f3460d5K2i
  18:     0x7fa61b6fd53d - trans::expr::trans_rvalue_dps_unadjusted::hd47de7ac66e018254zi
  19:     0x7fa61b6ae3f6 - trans::expr::trans_into::h95c6d2681fdd2548znh
  20:     0x7fa61b6af227 - trans::controlflow::trans_block::h3e86dfa8c58560e6b5d
  21:     0x7fa61b786821 - trans::base::trans_closure::hab3cc3c679d5ff23Kkt
  22:     0x7fa61b697b08 - trans::base::trans_fn::he0569b8eb832adf9Dvt
  23:     0x7fa61b6995a0 - trans::monomorphize::monomorphic_fn::hc1b7393dd1dc77f1usd
  24:     0x7fa61b6dee4e - trans::callee::trans_fn_ref_with_substs::hd01acb4398310d154kg
  25:     0x7fa61b6dd43e - trans::callee::trans_fn_ref::hb48e614c9b6dd9bcE9f
  26:     0x7fa61b6da88d - trans::callee::trans::ha56f4fe94448e6baVYf
  27:     0x7fa61b6f0fbb - trans::callee::trans_call_inner::h9722042290657949952
  28:     0x7fa61b6fd9e0 - trans::expr::trans_rvalue_dps_unadjusted::hd47de7ac66e018254zi
  29:     0x7fa61b6ae3f6 - trans::expr::trans_into::h95c6d2681fdd2548znh
  30:     0x7fa61b72e6f9 - trans::expr::trans_adt::h1af69b9b4e52152aO6i
  31:     0x7fa61b7311af - trans::expr::trans_struct::closure.42069
  32:     0x7fa61b71b016 - trans::expr::trans_struct::hcae8f9103f3460d5K2i
  33:     0x7fa61b6fd53d - trans::expr::trans_rvalue_dps_unadjusted::hd47de7ac66e018254zi
  34:     0x7fa61b6fb668 - trans::expr::trans_unadjusted::hfd3a5e1b5cbe37d5z4h
  35:     0x7fa61b6afb58 - trans::expr::trans::h23d7d0dd91a5190fHth
  36:     0x7fa61b6ec9f3 - trans::callee::trans_args::h29a92a6ed71c85ebm1g
  37:     0x7fa61b6f1ea0 - trans::callee::trans_call_inner::h9722042290657949952
  38:     0x7fa61b6fd9e0 - trans::expr::trans_rvalue_dps_unadjusted::hd47de7ac66e018254zi
  39:     0x7fa61b6fb668 - trans::expr::trans_unadjusted::hfd3a5e1b5cbe37d5z4h
  40:     0x7fa61b6afb58 - trans::expr::trans::h23d7d0dd91a5190fHth
  41:     0x7fa61b6adee3 - trans::expr::trans_into::h95c6d2681fdd2548znh
  42:     0x7fa61b6af227 - trans::controlflow::trans_block::h3e86dfa8c58560e6b5d
  43:     0x7fa61b6fcf7e - trans::expr::trans_rvalue_dps_unadjusted::hd47de7ac66e018254zi
  44:     0x7fa61b6ae3f6 - trans::expr::trans_into::h95c6d2681fdd2548znh
  45:     0x7fa61b6af227 - trans::controlflow::trans_block::h3e86dfa8c58560e6b5d
  46:     0x7fa61b786821 - trans::base::trans_closure::hab3cc3c679d5ff23Kkt
  47:     0x7fa61b72f139 - trans::closure::trans_closure_expr::hc99a1619c43ed333flx
  48:     0x7fa61b6fda38 - trans::expr::trans_rvalue_dps_unadjusted::hd47de7ac66e018254zi
  49:     0x7fa61b6fb764 - trans::expr::trans_unadjusted::hfd3a5e1b5cbe37d5z4h
  50:     0x7fa61b6afb58 - trans::expr::trans::h23d7d0dd91a5190fHth
  51:     0x7fa61b6ec9f3 - trans::callee::trans_args::h29a92a6ed71c85ebm1g
  52:     0x7fa61b6f1ea0 - trans::callee::trans_call_inner::h9722042290657949952
  53:     0x7fa61b6fd9e0 - trans::expr::trans_rvalue_dps_unadjusted::hd47de7ac66e018254zi
  54:     0x7fa61b6ae3f6 - trans::expr::trans_into::h95c6d2681fdd2548znh
  55:     0x7fa61b6af227 - trans::controlflow::trans_block::h3e86dfa8c58560e6b5d
  56:     0x7fa61b786821 - trans::base::trans_closure::hab3cc3c679d5ff23Kkt
  57:     0x7fa61b697b08 - trans::base::trans_fn::he0569b8eb832adf9Dvt
  58:     0x7fa61b6938a1 - trans::base::trans_item::h48fc370b7d259ac7vTt
  59:     0x7fa61b78f7ec - trans::base::trans_crate::hc92be67ede893c70GPu
  60:     0x7fa61c672e83 - driver::phase_4_translate_to_llvm::h9904f5d5fc3fb761rNa
  61:     0x7fa61c64e83f - driver::compile_input::h3913ff7013f0c056Iba
  62:     0x7fa61c716cb7 - run_compiler::h28a4446bae1034e7H5b
  63:     0x7fa61c714829 - thunk::F.Invoke<A, R>::invoke::h6503055919709693733
  64:     0x7fa61c7134a0 - rt::unwind::try::try_fn::h1384674024000742916
  65:     0x7fa61c0d4de8 - rust_try_inner
  66:     0x7fa61c0d4dd5 - rust_try
  67:     0x7fa61c713c3f - thunk::F.Invoke<A, R>::invoke::h5780663349966142752
  68:     0x7fa61c04f965 - sys::thread::thread_start::h4ab695857833a5dar8E
  69:     0x7fa615ecc373 - start_thread
  70:     0x7fa61bc1a27c - __clone
  71:                0x0 - <unknown>

Could not compile `env_logger`.

To learn more, run the command again with --verbose.
╭─mindcat@mindcat-linux-pc ~/workspace/nekoneko  ‹master*› 
╰─➤  rustc --version                                                                                                                101 ↵
rustc 1.0.0-nightly (3b3bb0e68 2015-03-04) (built 2015-03-05)
```

**Comment by mindcat** [comment]
```
╭─mindcat@mindcat-linux-pc ~/workspace/nekoneko  ‹master*› 
╰─➤  rustc --version --verbose
rustc 1.0.0-nightly (3b3bb0e68 2015-03-04) (built 2015-03-05)
binary: rustc
commit-hash: 3b3bb0e682c2d252e9f62dd9df5cff9552af91ad
commit-date: 2015-03-04
build-date: 2015-03-05
host: x86_64-unknown-linux-gnu
release: 1.0.0-nightly
```

**Comment by alexcrichton** [comment]
cc @michaelwoerister, but perhaps this is a dupe? I've seen a few bugs like this recently

**Comment by blackbeam** [comment]
Similar for mysql was introduced with latest nightly `rustc 1.0.0-nightly (3b3bb0e68 2015-03-04) (built 2015-03-04)`:

```
/Users/rustbuild/src/rust-buildbot/slave/nightly-dist-rustc-mac/build/src/libstd/sync/mutex.rs:177:37: 180:2 error: internal compiler error: debuginfo: Could not find scope info for node NodeExpr(Expr { id: 99160, node: ExprStruct(Path { span: Span { lo: BytePos(4745586), hi: BytePos(4745597), expn_id: ExpnId(4294967295) }, global: false, segments: [PathSegment { identifier: StaticMutex#0, parameters: AngleBracketedParameters(AngleBracketedParameterData { lifetimes: [], types: [], bindings: [] }) }] }, [Field { ident: Spanned { node: lock#0, span: Span { lo: BytePos(1770102), hi: BytePos(1770106), expn_id: ExpnId(4294967295) } }, expr: Expr { id: 99161, node: ExprPath(None, Path { span: Span { lo: BytePos(4745610), hi: BytePos(4745625), expn_id: ExpnId(4294967295) }, global: false, segments: [PathSegment { identifier: sys#0, parameters: AngleBracketedParameters(AngleBracketedParameterData { lifetimes: [], types: [], bindings: [] }) }, PathSegment { identifier: MUTEX_INIT#0, parameters: AngleBracketedParameters(AngleBracketedParameterData { lifetimes: [], types: [], bindings: [] }) }] }), span: Span { lo: BytePos(4745610), hi: BytePos(4745625), expn_id: ExpnId(4294967295) } }, span: Span { lo: BytePos(4745604), hi: BytePos(4745625), expn_id: ExpnId(4294967295) } }, Field { ident: Spanned { node: poison#0, span: Span { lo: BytePos(1770129), hi: BytePos(1770135), expn_id: ExpnId(4294967295) } }, expr: Expr { id: 99162, node: ExprPath(None, Path { span: Span { lo: BytePos(4745639), hi: BytePos(4745656), expn_id: ExpnId(4294967295) }, global: false, segments: [PathSegment { identifier: poison#0, parameters: AngleBracketedParameters(AngleBracketedParameterData { lifetimes: [], types: [], bindings: [] }) }, PathSegment { identifier: FLAG_INIT#0, parameters: AngleBracketedParameters(AngleBracketedParameterData { lifetimes: [], types: [], bindings: [] }) }] }), span: Span { lo: BytePos(4745639), hi: BytePos(4745656), expn_id: ExpnId(4294967295) } }, span: Span { lo: BytePos(4745631), hi: BytePos(4745656), expn_id: ExpnId(4294967295) } }], None), span: Span { lo: BytePos(4745586), hi: BytePos(4745659), expn_id: ExpnId(4294967295) } })
note: the compiler unexpectedly panicked. this is a bug.
note: we would appreciate a bug report: https://github.com/rust-lang/rust/blob/master/CONTRIBUTING.md#bug-reports
note: run with `RUST_BACKTRACE=1` for a backtrace
thread 'rustc' panicked at 'Box<Any>', /Users/rustbuild/src/rust-buildbot/slave/nightly-dist-rustc-mac/build/src/libsyntax/diagnostic.rs:129

stack backtrace:
   1:        0x108261b52 - sys::backtrace::write::h6b0889bc971c1ad3IDA
   2:        0x108290844 - panicking::on_panic::h0fcd1d74630df38dKsJ
   3:        0x1081aec67 - rt::unwind::begin_unwind_inner::h56b969a14fc61916gbJ
   4:        0x10793b95e - rt::unwind::begin_unwind::h7097931347514503270
   5:        0x10793b90b - diagnostic::SpanHandler::span_bug::hd809ae68b3d05ee6h0D
   6:        0x105110d7c - session::Session::span_bug::h5bc4aa3cff230a98ISp
   7:        0x104e34134 - trans::debuginfo::scope_metadata::hbc8249b53299cdbdSjE
   8:        0x104d3efb9 - trans::debuginfo::set_source_location::h27a07f90a8aeec578MD
   9:        0x104cf0925 - trans::expr::trans_into::h4a29f680db4b3526znh
  10:        0x104cf0c45 - trans::expr::trans_into::h4a29f680db4b3526znh
  11:        0x104d572c0 - trans::expr::trans_uniq_expr::h9953b7ba5686a321ukj
  12:        0x104d58053 - trans::expr::trans_unary::h112ae451f6aff215Jgj
  13:        0x104d416fc - trans::expr::trans_unadjusted::h66feeb4ce59b408dz4h
  14:        0x104cf0f79 - trans::expr::trans_into::h4a29f680db4b3526znh
  15:        0x104d76220 - trans::expr::trans_adt::h73a11f7497464a42O6i
  16:        0x104d78d52 - trans::expr::trans_struct::closure.42052
  17:        0x104d623cb - trans::expr::trans_struct::h4294120fb3439252K2i
  18:        0x104d436be - trans::expr::trans_rvalue_dps_unadjusted::hca2c07ae951484c14zi
  19:        0x104cf0f59 - trans::expr::trans_into::h4a29f680db4b3526znh
  20:        0x104cf1e29 - trans::controlflow::trans_block::h26a308528bb95051b5d
  21:        0x104dd0da9 - trans::base::trans_closure::h1e78b4450eb13a03Kkt
  22:        0x104cd9645 - trans::base::trans_fn::h31c9353b85ff9dd9Dvt
  23:        0x104cdb1c3 - trans::monomorphize::monomorphic_fn::h8c4abbed98c102adusd
  24:        0x104d2358b - trans::callee::trans_fn_ref_with_substs::heb22ab8904f67bf64kg
  25:        0x104d21b0e - trans::callee::trans_fn_ref::hb2be9d8aec49d9c4E9f
  26:        0x104d1f1d0 - trans::callee::trans::ha12ccf10898e9b03VYf
  27:        0x104d3624d - trans::callee::trans_call_inner::h11221320451677754653
  28:        0x104d43bec - trans::expr::trans_rvalue_dps_unadjusted::hca2c07ae951484c14zi
  29:        0x104d4151c - trans::expr::trans_unadjusted::h66feeb4ce59b408dz4h
  30:        0x104cf287b - trans::expr::trans::h60a968a47a19defeHth
  31:        0x104d317e1 - trans::callee::trans_args::h43fd254006ea96e2m1g
  32:        0x104d3734e - trans::callee::trans_call_inner::h11221320451677754653
  33:        0x104d43bec - trans::expr::trans_rvalue_dps_unadjusted::hca2c07ae951484c14zi
  34:        0x104cf0f59 - trans::expr::trans_into::h4a29f680db4b3526znh
  35:        0x104d76220 - trans::expr::trans_adt::h73a11f7497464a42O6i
  36:        0x104d78d52 - trans::expr::trans_struct::closure.42052
  37:        0x104d623cb - trans::expr::trans_struct::h4294120fb3439252K2i
  38:        0x104d436be - trans::expr::trans_rvalue_dps_unadjusted::hca2c07ae951484c14zi
  39:        0x104cf0f59 - trans::expr::trans_into::h4a29f680db4b3526znh
  40:        0x104d76220 - trans::expr::trans_adt::h73a11f7497464a42O6i
  41:        0x104d30545 - trans::base::trans_named_tuple_constructor::h1ca92235818072b2czt
  42:        0x104d366d1 - trans::callee::trans_call_inner::h11221320451677754653
  43:        0x104d43bec - trans::expr::trans_rvalue_dps_unadjusted::hca2c07ae951484c14zi
  44:        0x104cf0f59 - trans::expr::trans_into::h4a29f680db4b3526znh
  45:        0x104cf1e29 - trans::controlflow::trans_block::h26a308528bb95051b5d
  46:        0x104dd0da9 - trans::base::trans_closure::h1e78b4450eb13a03Kkt
  47:        0x104cd9645 - trans::base::trans_fn::h31c9353b85ff9dd9Dvt
  48:        0x104cd5aa6 - trans::base::trans_item::h8ce277cc13b040a3vTt
  49:        0x104cd58a7 - trans::base::trans_item::h8ce277cc13b040a3vTt
  50:        0x104cd58a7 - trans::base::trans_item::h8ce277cc13b040a3vTt
  51:        0x104dd9ecb - trans::base::trans_crate::hb69c466e59630a14GPu
  52:        0x10470eec7 - driver::phase_4_translate_to_llvm::he4e045cdc3897b4crNa
  53:        0x1046ea7f2 - driver::compile_input::h25d96a14ec26932cIba
  54:        0x1047bc72e - run_compiler::h15c3ea085a111a6fH5b
  55:        0x1047b9ac7 - thunk::F.Invoke<A, R>::invoke::h11905936354613679413
  56:        0x1047b860f - rt::unwind::try::try_fn::h4025213131938170097
  57:        0x10830d628 - rust_try_inner
  58:        0x10830d615 - rust_try
  59:        0x1047b8e48 - thunk::F.Invoke<A, R>::invoke::h13725635537516470714
  60:        0x1082790a2 - sys::thread::thread_start::h1120d6d2e105321012E
  61:     0x7fff8ab44267 - _pthread_body
  62:     0x7fff8ab441e4 - _pthread_start

Could not compile `mysql`.
```

**Comment by viperscape** [comment]
[Same(?) issue](https://www.refheap.com/98138) since newest rustc updated; I do not make use of Box<Any>, but I do have a mutex I use in [rust-promise](https://github.com/viperscape/rust-promise), which is what's failing to compile now. FWIW this compiled on Windows fine, but on Linux I receive this ICE

**Comment by golddranks** [comment]
And seems to be the same as https://github.com/rust-lang/rust/issues/23053 and https://github.com/rust-lang/rust/issues/23093

**Comment by golddranks** [comment]
Btw. I'm having the same ICE with OS X `rustc 1.0.0-nightly (3b3bb0e68 2015-03-04) (built 2015-03-04)` nightly when compiling `openssl-sys v0.5.1`.

**Comment by tshakah** [comment]
I'm also having the same problem trying to compile openssl-sys 0.5.1

**Comment by ninjabear** [comment]
I'm having the same problem as @tshakah, openssl-sys 0.5.1, slightly different nightly though ```rustc 1.0.0-nightly (b0746ff19 2015-03-05) (built 2015-03-06)```, same version of ubuntu

**Comment by aatxe** [comment]
I'm having the exact same issue in my IRC library with openssl-sys 0.5.1. When compiling without it, the same ICE happens in my own code.

**Comment by michaelwoerister** [comment]
:police_car: Woh! Here comes the dupe-police! :police_car: :police_car:
This is a duplicate of #22432, a fix is already in the queue: #23074
Thanks for reporting! Keep your eyes open, citizens!
